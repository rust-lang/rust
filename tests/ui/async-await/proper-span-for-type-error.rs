//@ edition:2021
//@ run-rustfix
#![allow(dead_code)]

async fn a() {}

async fn foo() -> Result<(), i32> {
    a().await //~ ERROR mismatched types
}

fn main() {}
