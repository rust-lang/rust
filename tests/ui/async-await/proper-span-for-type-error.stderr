error[E0308]: mismatched types
  --> $DIR/proper-span-for-type-error.rs:8:5
   |
LL |     a().await
   |     ^^^^^^^^^ expected `Result<(), i32>`, found `()`
   |
   = note:   expected enum `Result<(), i32>`
           found unit type `()`
help: try adding an expression at the end of the block
   |
LL ~     a().await;
LL ~     Ok(())
   |

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0308`.
